src/System.Net.Requests/src/System/Net/_TimerThread.cs

```diff
@@ -0,0 +1,873 @@
+                            {
+                                GlobalLog.Print("TimerThread#" + Thread.CurrentThread.ManagedThreadId.ToString() + "::ThreadProc() Waiting for " + waitDuration + "ms");
+                            }
+
+                            int waitResult = WaitHandle.WaitAny(s_ThreadEvents, waitDuration, false);
+
+                            // 0 is s_ThreadShutdownEvent - die.
+                            if (waitResult == 0) {
+                                if (GlobalLog.IsEnabled)
+                                {
+                                    GlobalLog.Print("TimerThread#" + Thread.CurrentThread.ManagedThreadId.ToString() + "::ThreadProc() Awoke, cause: Shutdown");
+                                }
+                                running = false;
+                                break;
+                            }
+
+                            if (GlobalLog.IsEnabled)
+                            {
+                                GlobalLog.Print("TimerThread#" + Thread.CurrentThread.ManagedThreadId.ToString() + "::ThreadProc() Awoke, cause: " + (waitResult == WaitHandle.WaitTimeout ? "Timeout" : "Prod"));
+                            }
+
+                            // If we timed out with nothing to do, shut down. 
+                            if (waitResult == WaitHandle.WaitTimeout && !haveNextTick) {
+                                Interlocked.CompareExchange(ref s_ThreadState, (int) TimerThreadState.Idle, (int) TimerThreadState.Running);
+                                // There could have been one more prod between the wait and the exchange.  Check, and abort if necessary.
+                                if (s_ThreadReadyEvent.WaitOne(0, false)) {
+                                    if (Interlocked.CompareExchange(ref s_ThreadState, (int) TimerThreadState.Running, (int) TimerThreadState.Idle) ==
+                                        (int) TimerThreadState.Idle) {
+                                        continue;
+                                    }
+                                }
+
+                                running = false;
+                                break;
+                            }
+                        }
+                    }
+                    catch (Exception exception) {
+                        if (ExceptionCheck.IsFatal(exception)) throw;
+
+#if FEATURE_FTP_LOGGING
+                        if (Logging.On) Logging.PrintError(Logging.Web, "TimerThread#" + Thread.CurrentThread.ManagedThreadId.ToString(NumberFormatInfo.InvariantInfo) + "::ThreadProc() - Exception:" + exception.ToString());
+#endif
+                        if (GlobalLog.IsEnabled)
+                        {
+                            GlobalLog.Print("TimerThread#" + Thread.CurrentThread.ManagedThreadId.ToString() + "::ThreadProc() exception: " + exception);
+                        }
+
+                        // The only options are to continue processing and likely enter an error-loop,
```

Comment:
/cc @ericeil 
I doubt this is true anymore: I remember that certain fatal exceptions cannot be caught in CoreFX. Not sure if we're bringing that behavior back.